File: typescript/base-hmi-types/src/screens/base/HmiTrendOverlay.ts
import { HmiProperty } from "./HmiProperty.js";
import { HmiTrendOverlayAnchorType } from "./HmiTrendOverlayAnchorType.js";

export class HmiTrendOverlay {
  snapshotName?: string;
  anchorType?: HmiProperty<HmiTrendOverlayAnchorType>;
  startTime?: string;
  useDefaultStartTime?: HmiProperty<boolean>;
  offsetPercent?: HmiProperty<number>;
  visible?: HmiProperty<boolean>;
  startPenNumber?: HmiProperty<number>;
  endPenNumber?: HmiProperty<number>;
}

File: src/BaseHmiTypes/Screens/Base/HmiTrendOverlay.cs
namespace BaseHmiTypes.Screens.Base;

public sealed class HmiTrendOverlay
{
    public string? SnapshotName { get; set; }

    public HmiProperty<HmiTrendOverlayAnchorType>? AnchorType { get; set; }

    public string? StartTime { get; set; }

    public HmiProperty<bool>? UseDefaultStartTime { get; set; }

    public HmiProperty<double>? OffsetPercent { get; set; }

    public HmiProperty<bool>? Visible { get; set; }

    public HmiProperty<int>? StartPenNumber { get; set; }

    public HmiProperty<int>? EndPenNumber { get; set; }
}
